llvm/lib/Transforms/Vectorize/VPlanConstantFolder.h

```diff
@@ -0,0 +1,98 @@
+//===- VPlanConstantFolder.h - ConstantFolder for VPlan -------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "VPlanValue.h"
+#include "llvm/IR/ConstantFolder.h"
+
+#ifndef LLVM_TRANSFORMS_VECTORIZE_VPLANCONSTANTFOLDER_H
+#define LLVM_TRANSFORMS_VECTORIZE_VPLANCONSTANTFOLDER_H
+
+namespace llvm {
+class VPConstantFolder {
+private:
+  ConstantFolder Folder;
```

Comment:
It is better to use TargetFolder instead of ConstantFolder.

Reply:
I know, but the DataLayout is not readily available at the VPlan level, and I wonder if it makes sense to thread it into VPlan?

Reply:
If you want to do constant folding, then yes, you should thread it into VPlan.

Reply:
Thanks! There are lots of test changes now, and I wonder if the patch should be split out.